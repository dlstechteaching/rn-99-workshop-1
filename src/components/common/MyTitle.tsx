import React from "react";
import { Text, StyleSheet, View } from "react-native";

interface MyTitleProps {
    label: string;
}

export const MyTitle: React.VFC<MyTitleProps> = ({ label }) => {
    return (
        <View style={styles.view}>
            <Text style={styles.text}>{ label }</Text>
        </View>
    );

};

const styles = StyleSheet.create({
    view: {
        marginVertical: 50
    },
    text: {
        color: 'black',
        textAlign: 'center',
        fontSize: 35,
        fontWeight: 'bold',
    }
});