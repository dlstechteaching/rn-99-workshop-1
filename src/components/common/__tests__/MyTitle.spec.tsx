import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { MyTitle } from "../MyTitle";

describe("MyTitle tests", () => {
  let component: ShallowWrapper;

  it("should render properly", () => {
    givenComponent();
    thenItRendersProperly();
  });

  function givenComponent() {
    component = shallow(
      <MyTitle label={"label"} />
    );
  }

  function thenItRendersProperly() {
    expect(component).toMatchSnapshot();
  }

});