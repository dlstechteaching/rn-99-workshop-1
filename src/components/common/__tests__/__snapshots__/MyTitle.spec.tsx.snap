// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyTitle tests should render properly 1`] = `
<View
  style={
    Object {
      "marginVertical": 50,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 35,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    label
  </Text>
</View>
`;
